rust/main/agents/relayer/src/server/message_retry.rs

```diff
@@ -1,32 +1,95 @@
 use crate::settings::matching_list::MatchingList;
+
 use axum::{extract::State, routing, Json, Router};
+
 use derive_new::new;
-use tokio::sync::broadcast::Sender;
+use serde::{Deserialize, Serialize};
+use tokio::sync::{broadcast::Sender, mpsc};
 
 const MESSAGE_RETRY_API_BASE: &str = "/message_retry";
 
-#[derive(new, Clone)]
+#[derive(new)]
 pub struct MessageRetryApi {
-    tx: Sender<MatchingList>,
+    retry_request_transmitter: Sender<MessageRetryRequest>,
+}
+
+#[derive(Clone, Debug)]
+pub struct MessageRetryApiState {
```

Comment:
what's the difference between this and `MessageRetryApi` now? can we consolidate into one type?

Reply:
It'd have to be moved to `hyperlane-base` to be imported in both usage sites, but this is a relayer-specific data structure so may not fit that well in `-base`. I think we have similar duplication between agent code and e2e elsewhere too, cleanest would be to define a new crate where these types are defined but I personally was ok with this small duplication (CI would fail if the structs diverged)

Reply:
It's so we can create the router.

```rust
impl MessageRetryApi {
    pub fn router(&self) -> Router {
        Router::new()
            .route("/", routing::post(retry_message))
            .with_state(MessageRetryApiState {
                retry_request_transmitter: self.retry_request_transmitter.clone(),
            })
    }
```

But yeah, we can totally consolidate it